### ISSUE TITLE
Disallow hiding difficulty of beatmap with only one visible difficulty

### ISSUE THREAD

**Issue by ghost** [ppy/osu#20636]
### Type

Game behaviour

### Bug description

Currently it is possible to rightclick the difficulty of a beatmap with only a single non-hidden difficulty and click "Hide", which hides not only that difficulty but the whole beatmap from the list, making it impossible to re-retrieve that beatmap in order to unhide difficulties.

I now have 2 broken beatmaps in my list that I can't see or play because I hid all of their difficulties. This means I can't use "Restore all hidden" either. They don't show up if I search for their names or if I try to navigate to them from the beatmap listing. It shows no results.

### Version

2022.1008.2-lazer

### Logs

None.

**Comment by nihodi** [comment]
Can't you use the "Restore all hidden difficulties" button in the Maintenance section in the settings?

**Comment by r00ster91** [comment]
Yup, that works. I still think this being allowed is confusing behavior however and should be disallowed. What purpose does it serve? If you don't want to see the beatmap at all (by hiding all of its difficulties), you might as well delete it. So I think it should still be disallowed.

**Comment by nihodi** [comment]
Take it to discussions maybe?

**Comment by Feodor0090** [comment]
There are some maps those i sometimes play with my friends in multi, but i don't want to see them in carousel. Not to redownload -> delete each time i just hide them. Instead of restricting this feature, i want to request an ability to view hidden maps in maintenance settings somehow and ability to restore them one-by-one. Also would be nice to have "hide all difficulties" in BMS's context menu. Painful to hide them one-by-one if there are 8 or 10 of them.

**Comment by loma-baldsson** [comment]
I think hiding beatmaps should send them to some sort of "hidden" collection, so restoring them could be as easy as tabbing into that collection and moving it out.
